src/lib/modules/pipeline/webpackProcess.js

```diff
@@ -48,38 +48,42 @@ class WebpackProcess extends ProcessWrapper {
       }
 
       if (typeof config !== 'object' || config === null) {
-        return callback('bad webpack config, the resolved config was null or not an object');
+        return callback('Pipeline: '.cyan + 'bad webpack config, the resolved config was null or not an object');
       }
 
       webpack(config).run(async (err, stats) => {
         if (err) {
           return callback(errorMessage(err));
         }
-        if (!config.stats || config.stats === 'none') {
-          return callback();
-        }
-        try {
-          this._log('info', 'writing file ' + ('.embark/stats.report').bold.dim);
-          await writeFile(
-            fs.dappPath('.embark/stats.report'),
-            stats.toString(config.stats)
-          );
-          this._log('info', 'writing file ' + ('.embark/stats.json').bold.dim);
-          await writeFile(
-            fs.dappPath('.embark/stats.json'),
-            JSON.stringify(stats.toJson(config.stats))
-          );
-          if (stats.hasErrors()) {
-            const errors = stats.toJson(config.stats).errors.join('\n');
-            return callback(errors);
-          }
-          callback();
-        } catch (e) {
-          return callback(errorMessage(e));
-        }
+        callback(null, config, stats);
       });
     });
   }
+
+  async writeStats(config, stats, callback){
+    if (!config.stats || config.stats === 'none') {
+      return callback();
+    }
+    try {
+      this._log('info', 'Pipeline: '.cyan + 'writing file ' + ('.embark/stats.report').bold.dim);
```

Comment:
No need to call `_log` yourself. The ProcessWrapper catches `console.x` and calls `_log` for you. So you can call `console.info` in that case.
Also, using string templates would be cleaner.

Reply:
I think @michaelsbradleyjr was filtering out some hardsource plugin logs using `this._log`.